config/class-site-details-index.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+namespace Automattic\VIP\Config;
+
+class Site_Details_Index {
+	private static $_instance;
+
+	private $timestamp = null;
+
+	/**
+	 * Standard singleton except accept a timestamp for mocking purposes
+	 *
+	 * @param {mixed} $timestamp A fixed point in time to use for mocking
+	 * @return {Site_Details_Index} A Site_Details_Index object
+	 */
+	public static function instance( $timestamp = null ) {
+		if ( ! ( static::$_instance instanceof Site_Details_Index ) ) {
+			static::$_instance = new Site_Details_Index();
+			static::$_instance->set_current_timestamp( $timestamp );
+			static::$_instance->init();
+		}
+
+		return static::$_instance;
+	}
+
+	public function init() {
+		add_filter( 'vip_site_details_index_data', [ $this, 'set_env_and_core' ] );
+	}
+
+	/**
+	 * Given the site details from the vip_site_details_index_data filter, set the environment and core values
+	 *
+	 * @param {array} $site_details The current site details
+	 * @return {array} The new site details
+	 */
+	public function set_env_and_core( $site_details ) {
+		global $wp_version;
+
+		$site_id = 0;
+		if ( defined( 'VIP_GO_APP_ID' ) && VIP_GO_APP_ID ) {
```

Comment:
SImilarly to the below, feels like using either `VIP_GO_APP_ID` or `FILES_CLIENT_SITE_ID` is sufficient, since they both use the same value on config generation and there wouldn't be a case where one was defined and the other wouldn't.